src/libraries/System.Private.CoreLib/src/System/DateTime.cs

```diff
@@ -1351,92 +1352,58 @@ public DateTime Date
-            // estimate for the month
-            uint m = (n >> 5) + 1;
-            // m = 1-based month number
-            while (n >= days[m]) m++;
-            // If month was requested, return it
-            if (part == DatePartMonth) return (int)m;
-            // Return 1-based day-of-month
-            return (int)(n - days[m - 1] + 1);
         }
 
         // Exactly the same as GetDatePart, except computing all of
         // year/month/day rather than just one of them. Used when all three
         // are needed rather than redoing the computations for each.
         internal void GetDate(out int year, out int month, out int day)
         {
-            // n = number of days since 1/1/0001
-            uint n = (uint)(UTicks / TicksPerDay);
-            // y400 = number of whole 400-year periods since 1/1/0001
-            uint y400 = n / DaysPer400Years;
-            // n = day number within 400-year period
-            n -= y400 * DaysPer400Years;
-            // y100 = number of whole 100-year periods within 400-year period
-            uint y100 = n / DaysPer100Years;
-            // Last 100-year period has an extra day, so decrement result if 4
-            if (y100 == 4) y100 = 3;
-            // n = day number within 100-year period
-            n -= y100 * DaysPer100Years;
-            // y4 = number of whole 4-year periods within 100-year period
-            uint y4 = n / DaysPer4Years;
-            // n = day number within 4-year period
-            n -= y4 * DaysPer4Years;
-            // y1 = number of whole years within 4-year period
-            uint y1 = n / DaysPerYear;
-            // Last year has an extra day, so decrement result if 4
-            if (y1 == 4) y1 = 3;
-            // compute year
-            year = (int)(y400 * 400 + y100 * 100 + y4 * 4 + y1 + 1);
-            // n = day number within year
-            n -= y1 * DaysPerYear;
-            // dayOfYear = n + 1;
-            // Leap year calculation looks different from IsLeapYear since y1, y4,
-            // and y100 are relative to year 1, not year 0
-            uint[] days = y1 == 3 && (y4 != 24 || y100 == 3) ? s_daysToMonth366 : s_daysToMonth365;
-            // All months have less than 32 days, so n >> 5 is a good conservative
-            // estimate for the month
-            uint m = (n >> 5) + 1;
-            // m = 1-based month number
-            while (n >= days[m]) m++;
-            // compute month and day
-            month = (int)m;
-            day = (int)(n - days[m - 1] + 1);
+            // y400 = number of whole 400-year periods since 3/1/0000
+            // r1 = day number within 400-year period
+            (uint y400, uint r1) = Math.DivRem(((uint)(UTicks / TicksPer6Hours) | 3U) + 1224, DaysPer400Years);
+            ulong u2 = (ulong)Math.BigMul(2939745, (int)r1 | 3);
+            ushort daySinceMarch1 = (ushort)((uint)u2 / 11758980);
+            var n3 = 2141 * daySinceMarch1 + 197913;
```

Comment:
Nit: Our [coding conventions](https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/coding-style.md) disallow vars, except in a few limited cases. These should be explicit types.

Reply:
fixed it